content/en/docs/tasks/administer-cluster/encrypt-data.md

```diff
@@ -65,72 +124,161 @@ resources:
+to how you would configure KMS encryption; however, the specific details of that depend
+on the (external) KMS integration that you are using.
 
-Encrypting secret data with a locally managed key protects against an etcd compromise, but it fails to
-protect against a host compromise. Since the encryption keys are stored on the host in the
-EncryptionConfiguration YAML file, a skilled attacker can access that file and extract the encryption
-keys.
+### Generate the encryption key
 
-Envelope encryption creates dependence on a separate key, not stored in Kubernetes. In this case,
-an attacker would need to compromise etcd, the `kubeapi-server`, and the third-party KMS provider to
-retrieve the plaintext values, providing a higher level of security than locally stored encryption keys.
+Start by generating a new encryption key, and then encode it using base64:
+
+{{< tabs name="generate_encryption_key" >}}
+{{% tab name="Linux" %}}
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```shell
+head -c 32 /dev/urandom | base64
+```
+
+You can use `/dev/hwrng` instead of `/dev/urandom` if you want to
+use your PC's built-in hardware entropy source. Not all Linux
+devices provide a hardware random generator.
+{{% /tab %}}
+{{% tab name="macOS" %}}
+<!-- localization note: this is similar to the Linux tab and the wording
+should match wherever the English text does -->
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```shell
+head -c 32 /dev/urandom | base64
+```
+{{% /tab %}}
+{{% tab name="Windows" %}}
+Generate a 32-byte random key and base64 encode it. You can use this command:
+```powershell
+# Do not run this in a session where you have set a random number
+# generator seed.
+[Convert]::ToBase64String((1..32|%{[byte](Get-Random -Max 256)}))
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+
+{{< note >}}
+Keep the encryption secret confidential, including whilst you generate it and
+ideally even after you are no longer actively using it.
+{{< /note >}}
+
+
+### Replicate the encryption key
 
-## Encrypting your data
+Using a secure mechanism for file transfer, make a copy of that encryption key
+available to every other control plane host.
 
-Create a new encryption config file:
+At a minimum, use encryption in transit - for example, secure shell (SSH). For more
+security, use asymmetric encryption between hosts, or change the approach you are using
+so that you're relying on KMS encryption.
+
+### Write an encryption configuration file
+
+{{< caution >}}
+The encryption configuration file may contain keys that can decrypt content in etcd.
+If the configuration file contains any key material, you must properly
+restrict permissions on all your control plane hosts so only the user
+who runs the kube-apiserver can read this configuration.
+
+You may need to create a secure backup of the configuration file. If so, do
```

Comment:
Which "configuration file" should I backup? The encryption configuration file is yet to be written.